src/bson_mapper/bson_archiver.hpp

```diff
@@ -45,6 +45,7 @@
 
 #include <chrono>
 #include <memory>
+#include <sstream>
```

Comment:
Where exactly is the error? Did you try getting rid of it here and putting it in demo.cpp?